pages/api/moralis/pinFile.ts

```diff
@@ -9,13 +10,34 @@ export const config = {
   },
 };
 
+const toBase64 = (file: ArrayBuffer) => {
+    return Buffer.from(file).toString('base64');
+  };
+
 export default async function handler(
   request: NextApiRequest,
   response: NextApiResponse
 ) {
-  const { b64File, name } = request.body;
+  const { url, name } = request.body;
+
+  const fileResponse = await fetch(url);
+  const fileResult = await fileResponse.arrayBuffer();
+
+  let b64File = toBase64(fileResult);
+
+  try {
+    const deleteFiles = await supabase
+      .storage
+      .from('audio-files')
+      .remove([name])
+
+    if (deleteFiles.error) throw deleteFiles.error;
+  } catch (e) {
+    console.error(e);
+  }
+
   try {
-    const res = await pinFile(b64File as string, name as string);
+    const res = await pinFile(b64File as string, 'phlotexyzfile' as string);
```

Comment:
You have the `File`, just use the `file.name`. I think that'd be better.

Reply:
I actually tried passing file.name but Moralis doesn't support some character's that can be present in a file name. I could clean the file name before sending, but since it was undefined previously I didn't think it was important.

Reply:
That's right we ran into this! Okay then it's unimportant